documenteer/sphinxconfig/utils.py

```diff
@@ -29,10 +29,152 @@ def read_git_branch():
+    ------
+    IOError
+        Raised if the ``filepath`` is set, but does not exist in the Git
+        repository.
+    """
+    repo = git.repo.base.Repo(path=repo_path, search_parent_directories=True)
+    head_commit = repo.head.commit
+
+    # most recent commit datetime of the given file
+    for commit in head_commit.iter_parents(filepath):
+        return commit.committed_datetime
+
+    # Only get here if git could not find the file path in the history
+    raise IOError('File {0} not found'.format(filepath))
+
+
+def get_filepaths_with_extension(extname, root_dir='.'):
+    """Get relative filepaths of files in a directory, and sub-directories,
+    with the given extension.
+
+    Parameters
+    ----------
+    extname : `str`
+        Extension name (e.g. 'txt', 'rst').
+    root_dir : 'str`, optional
+        Root directory. Current working directory by default.
+
+    Return
+    ------
+    filepaths : `list` of `str`
+        File paths, relative to ``root_dir``, with the given extension.
+    """
+    if not extname.startswith('.'):
+        extname = '.' + extname
+
+    root_dir = os.path.abspath(root_dir)
+
+    selected_filenames = []
+    for dirname, sub_dirnames, filenames in os.walk(root_dir):
+        for filename in filenames:
+            if os.path.splitext(filename)[-1].lower() == extname:
+                full_filename = os.path.join(dirname, filename)
+                selected_filenames.append(
+                    os.path.relpath(full_filename, start=root_dir))
+    return selected_filenames
+
+
+def get_project_content_commit_date(root_dir='.'):
+    """Get the datetime for the most recent commit to a project that
+    affected Sphinx content.
+
+    *Content* is considered any file with an ``rst``, ``png``, ``jpeg``,
+    ``jpg``, ``svg``, or ``gif`` extension. ``README.rst`` is excluded.
+
+    This function allows project infrastructure and configuration files to be
+    updated without changing the timestamp.
+
+    Parameters
+    ----------
+    root_dir : 'str`, optional
+        Root directory. Current working directory by default.
+
+    Returns
+    -------
+    commit_date : `datetime.datetime`
+        Datetime of the most recent content commit.
+
+    Raises
+    ------
+    RuntimeError
+        Raised if no content files are found.
+    """
+    content_paths = []
+
+    # Supported 'content' extensions
+    extensions = ('rst', 'png', 'jpeg', 'jpg', 'svg', 'gif')
+
+    for extname in extensions:
+        content_paths += get_filepaths_with_extension(
+            extname,
+            root_dir=root_dir)
+
+    # Known files that should be excluded; lower case for comparison
+    exclusions = ('readme.rst', 'license.rst')
+    # filter out excluded files
+    content_paths = [p for p in content_paths
+                     if p.lower() not in exclusions]
+
+    if len(content_paths) == 0:
+        raise RuntimeError('No content files found in {0}'.format(root_dir))
+
+    print(content_paths)
```

Comment:
If this is meant to appear in the build log, maybe you can add a descriptive sentence before it saying what is being printed?